apps/aecore/lib/aecore/sync/chain.ex

```diff
@@ -0,0 +1,148 @@
+defmodule Aecore.Sync.Chain do
+  @moduledoc """
+  Implements all the functions regarding the Chain structure of the SyncTask
+  """
+
+  alias Aecore.Chain.Header
+  alias Aecore.Chain.BlockValidation
+  alias Aecore.Sync.Task
+  alias __MODULE__
+
+  @type peer_id :: pid()
+  @type chain_id :: reference()
+  @type height :: non_neg_integer()
+  @type hash :: binary()
```

Comment:
what hash is it? don't we have more specific typespecs for block_hash, tx_hash or whatever this should be somewhere?